src/container/oid_iv.c

```diff
@@ -130,17 +133,28 @@ oid_iv_ent_update(struct ds_iv_entry *ns_entry, struct ds_iv_key *iv_key,
 	d_rank_t		myrank = dss_self_rank();
 	int			rc;
 
+	if (src == NULL) {
+		D_DEBUG(DB_MD, "%u: ON UPDATE delete entry iv_entry %p\n", myrank, ns_entry);
+		ns_entry->iv_to_delete = 1;
+		return 0;
+	}
+
 	D_ASSERT(priv != NULL);
 	entry = ns_entry->iv_value.sg_iovs[0].iov_buf;
-	rc = ABT_mutex_trylock(entry->lock);
-	/** For retry requests, from _iv_op(), the lock may not be released in some cases. */
-	if (rc == ABT_ERR_MUTEX_LOCKED && entry->current_req != priv)
-		return -DER_BUSY;
+	oids  = src->sg_iovs[0].iov_buf;
 
-	entry->current_req = priv;
-	avail = &entry->rg;
+	rc = ABT_mutex_trylock(entry->lock);
+	if (rc == ABT_ERR_MUTEX_LOCKED) {
+		if (entry->current_req_rank == oids->req_rank &&
+		    entry->current_req_ptr == oids->req_ptr)
+			D_DEBUG(DB_MD, "%u: ON UPDATE - SAME REQ %p \n", myrank, src);
+		else
+			return -DER_BUSY;
```

Comment:
On the middle node, if both child X, Y forward request to A -
X -> A -> P
Y -> A -> P
Assume X's request first arrive and goes up to P, then Y's request arrives
In that case looks should make it wait first request's reply, but current code will return EBUSY to Y, and Y will retry the req. This causes some unnecessary RPC + retry, it is worse if there are many layers between X and A.
We probably should think more for it and refine it later.

Reply:
i thought with ebusy, the retry is just local on A.. so it will be:
Y->A(busy spin)->P
or am i wrong?

Reply:
I think that is incorrect.
If the iv update initiated on A, then will retry on A.
But in the example case, the IV update initiated on Y, then A returns EBUSY will cause it reply X's request as EBUSY, and then on X will retry the IV update.
The retry only on initiated node.

Reply:
whatever the flow is, This PR does not change the behavior for this. the same thing happens before, right?

Reply:
right, I think can land it first and consider how to refine it later.

Reply:
> I think that is incorrect. If the iv update initiated on A, then will retry on A. But in the example case, the IV update initiated on Y, then A returns EBUSY will cause it reply X's request as EBUSY, and then on X will retry the IV update that will be forwarded to A again. So if there are many layers between Y and A, it will need to go through every layers for reply + retry on Y + forward The retry only on initiated node.

if that is the case, yes it's very bad.. it's better to block on A rather than send the ebusy back. sigh. we should have caught this when this ebusy was introduced.